metricbeat/module/kibana/stats/stats.go

```diff
@@ -101,17 +131,35 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 // It returns the event which is then forward to the output. In case of an error, a
 // descriptive error must be returned.
 func (m *MetricSet) Fetch(r mb.ReporterV2) {
-	content, err := m.http.FetchContent()
+	intervalMs := m.Module().Config().Period.Nanoseconds() / 1000 / 1000
```

Comment:
Could we have this code inside the `x-pack` part as it's only used there I think? Perhaps we can find a way to only if that xPackEnabled check once and run instead also in fetchStats. Why not move this to line 151 with the fetchSettings?

Reply:
I'm not sure I fully understood your comment but I took a stab at it in 3afa1a2a7. Take a look and let me know what you think. Thanks!

Reply:
LGTM